### DISCUSSION TITLE
How to Subclass GraphicsLayoutWidget

### DISCUSSION THREAD

**Discussion by igor-martinovski** [pyqtgraph/pyqtgraph#1982] [Q&A] [answered]
Hi Everyone,
I am trying to create a small application that will allow me to plot csv files.
I'd like to subclass the GraphicsLayoutWidget to add more convenient add/remove methods and also to make it easier to access PlotItem within the GraphicsLayoutWidget.
I am also using QUiLoader to load a ui file.

The question has to do with the second arg passed to the __init__ function of the subclass.
In the code this is `arg2`. Code is shown below.

```python
import sys, pdb
import numpy as np
from PySide6 import QtCore, QtGui, QtWidgets
from PySide6.QtUiTools import QUiLoader
import pyqtgraph as pg


class wsGraphicsLayout(pg.GraphicsLayoutWidget):
    plots = []
    rowspan = 1
    colspan = 1
    
    def add_plot(self, **kargs):
        self.plots.append(self.addPlot(**kargs))

    def __init__(self, arg2):  # Why does __init__ get 2 arguments?
        print ('arg2:', arg2)  # The subclass doesnt initialize without arg2.
        super().__init__()

loader = QUiLoader()
loader.registerCustomWidget(wsGraphicsLayout)
app = QtWidgets.QApplication(sys.argv)
wsplot = loader.load("wsplot_main.ui", None)

wsplot.show()
# graphicsView is defined in QtDesigner, kinda confusing. How to improve?
PlotLayout = wsplot.graphicsView
PlotLayout.add_plot(col=1, row=1)
PlotLayout.add_plot(col=1, row=2)
PlotLayout.add_plot(col=1, row=3)

app.exec()
```
Is this the correct way to do this? Is there anything I need to do with arg2?
Running the above code shows that arg2 is a QWidget

```
>python wsplot2.py
qt.pysideplugin: Environment variable PYSIDE_DESIGNER_PLUGINS is not set, bailing out.
qt.pysideplugin: No instance of QPyDesignerCustomWidgetCollection was found.
qt.pysideplugin: No instance of QPyDesignerCustomWidgetCollection was found.
arg2: <PySide6.QtWidgets.QWidget(0x19e6dd85260, name="centralwidget") at 0x0000019E6EB5F4C0>
```
Thanks!

**Comment by j9ac9k** [comment]
I don't use qt designer, so I'm not really in a position to speak to this; but if I was working on this myself, I would just insert a breakpoint; and step through the initializer so you can better identify what that second argument is; but the "centralwidget" there are only a few places in the pyqtgraph repo, specifically some of the template files that set this value:

https://github.com/pyqtgraph/pyqtgraph/blob/6f0ffcbf8f1d497373a56f14624e18ae10a3e9a6/examples/VideoTemplate_pyside6.py#L27

**Comment by ixjlyons** [comment, marked as answer]
It's the parent widget -- I think the code generated by Qt Designer always passes the parent when constructing a widget in the UI. It's pretty common to see this for example:

```python
def __init__(self, parent=None):
    super().__init__(parent=parent)
    ...
```

or just:

```python
def __init__(self, *args, **kwargs):
    super().__init__(*args, **kwargs)
    ...
```

**Reply by igor-martinovski** [reply]
Thanks guys, good to know about the loader behavior. The second solution above seems the best as it is the most general.
